presentation/viewmodel/src/main/kotlin/com/captures2024/soongan/presentation/viewmodel/main/home/RegistrationPostViewModel.kt

```diff
@@ -0,0 +1,280 @@
+            val newValue: String,
+        ) : Intent
+
+        data object OnClickSubmit : Intent
+
+        data object OnClickCancelSubmitBottomSheet : Intent
+
+        data object OnClickConfirmSubmitBottomSheet : Intent
+    }
+
+    init {
+        intent(Intent.Init)
+    }
+
+    override fun createInitialState(savedStateHandle: SavedStateHandle): State = State(
+        currentMedia = null,
+        currentContestInfo = null,
+        title = AppConst.EMPTY_STRING,
+        maxInputLength = AppConst.Main.Home.MAX_INPUT_LENGTH,
+        isShowBackDialog = false,
+        isOpenSubmitBottomSheet = false,
+    )
+
+    override fun handleClientException(throwable: Throwable) {
+        analyticsHelper.e(throwable = throwable) { "state: $currentState" }
+    }
+
+    override fun handleIntent(intent: Intent) {
+        when (intent) {
+            is Intent.Init -> loadingLaunch { handleInit() }
+            is Intent.OpenMediaPicker -> handleOpenMediaPicker()
+            is Intent.InitMedia -> handleInitMedia(intent)
+            is Intent.OnClickBack -> handleOnClickBack()
+            is Intent.OnClickCancelBackDialog -> handleOnClickCancelBackDialog()
+            is Intent.OnClickConfirmBackDialog -> handleOnClickConfirmBackDialog()
+            is Intent.OnTitleValueChanged -> handleOnTitleValueChanged(intent)
+            is Intent.OnClickSubmit -> handleOnClickSubmit()
+            is Intent.OnClickCancelSubmitBottomSheet -> handleOnClickCancelSubmitBottomSheet()
+            is Intent.OnClickConfirmSubmitBottomSheet -> loadingLaunch { handleOnClickConfirmSubmitBottomSheet() }
+        }
+    }
+
+    private suspend fun handleInit() {
+        val rounds = getWeeklyContestInfoListUseCase
+            .invoke()
+            .getOrNull()
+
+        if (rounds == null || rounds.weeklyContestInfoList.isEmpty()) {
+            postSideEffect(Effect.NavigateToBack)
+            return
+        }
+
+        reduce {
+            copy(
+                currentContestInfo = rounds.weeklyContestInfoList.last(),
+            )
+        }
+    }
+
+    private fun handleOpenMediaPicker() {
+        postSideEffect(Effect.OpenMediaPicker)
+    }
+
+    private fun handleInitMedia(intent: Intent.InitMedia) {
+        val uri = intent.mediaUri
+
+        if (uri == null) {
+            postSideEffect(Effect.NavigateToBack)
+            return
+        }
+
+        reduce {
+            copy(
+                currentMedia = intent.mediaUri,
+            )
+        }
+    }
+
+    private fun handleOnClickBack() {
+        showBackDialog()
+    }
+
+    private fun handleOnClickCancelBackDialog() {
+        dismissBackDialog()
+    }
+
+    private fun handleOnClickConfirmBackDialog() {
+        dismissBackDialog()
+        postSideEffect(Effect.NavigateToBack)
+    }
+
+    private fun handleOnTitleValueChanged(intent: Intent.OnTitleValueChanged) {
+        val newValue = intent.newValue
+
+        if (newValue.length !in 0..15) {
```

Comment:
```suggestion
        if (newValue.length > AppConst.Main.Home.MAX_INPUT_LENGTH) {
```
string이라 음수 값이 없어서 in보다 처리가 빠를 것 같아요.

Reply:
IntRange contains 구현이 list contains가 아니라 if 문으로 condition 체크하는 거 일거에요

Reply:
아하 그렇네요 매직 넘버 교체로 정정하겠습니다 ..ㅎㅎ